### ISSUE TITLE
Add argument to hide warnings

### ISSUE THREAD

**Issue by skylord123** [qpdf/qpdf#232]
I am calling qpdf from a coldfusion application and if any warnings are thrown coldfusion assumes it encountered errors (because warnings are thrown to stderr). There is no way to get the exit code from coldfusion (which would make this an easy fix) so I had to build code to parse the error output and see if it's only warnings (which is a very ugly fix imo).

So do warnings need to be updated to throw to stdout or do we just need an argument that hides warnings?

TLDR: I only care about warnings if they lead to an error.

**Comment by jberkenbilt** [comment]
The library already has a `QPDF::setSuppressWarnings()` method. Adding a `--suppress-warnings` flag to the command-line tool would be trivial. I will add this for 8.2.0, which I hope to release within the next two weeks. Good suggestion.

**Comment by skylord123** [comment]
Awesome! I will keep an eye out for the new version and give it a go when it is live. Thanks for the quick response, appreciate it.

**Comment by abhishek-27** [comment]
Why `--no-warn` still throws a return code 3?

So now to suppress warnings when called using subprocess, I had to explicitly check for return code == 3, as the --no-warn was still throwing to an except CalledProcessError block. 

Is there any recommendations/suggestions to handle this?

**Comment by jberkenbilt** [comment]
@abhishek-27 Please open a new issue requesting an option to also have warnings not exit 3. You are not the first one to run into this. In fact, I ran into it myself recently. I should be able to get a solution to this into the next release. Unfortunately I will probably have to add yet another command-line option to avoid breaking compatibility.

**Comment by jberkenbilt** [comment]
@abhishek-27 qpdf 10.0.2, to be released very soon (in a few days at most), will have a `--warnings-exit-0` option which will cause qpdf to exit 0 with warnings. When combined with `--no-warn`, this effectively causes qpdf to completely ignore warnings.

**Comment by gronke** [comment]
> @Abhishek-27 qpdf 10.0.2, to be released very soon (in a few days at most), will have a `--warnings-exit-0` option which will cause qpdf to exit 0 with warnings. When combined with `--no-warn`, this effectively causes qpdf to completely ignore warnings.

The CLI option is correctly called `--warning-exit-code` (singular "warning"). 

https://github.com/qpdf/qpdf/blob/5057bfa642c7711f7b356ac32ffba76ccb09f1f5/libqpdf/qpdf/auto_job_help.hh#L45
